### DISCUSSION TITLE
About h2 server config `DEFAULT_SETTINGS_MAX_HEADER_LIST_SIZE`

### DISCUSSION THREAD

**Discussion by wisdom-yzh** [hyperium/hyper#3687] [Q&A]
**Version**
hyper 1.3.1
h2 0.4.5

**Description**
Why `DEFAULT_SETTINGS_MAX_HEADER_LIST_SIZE` for h2 protocol changed from 16MB to 16KB in hyper 1.3.1 ?

In h2 lib and earlier hyper, the default value is 16MB, while in hyper-1.3.1 it changed to 16KB. As a result, some http requests triggered from golang x/net/http2 package with long url or big headers encounter failure `http2: request header list larger than peer's advertised limit`.

**Comment by seanmonstar** [comment]
The default changed to something safer, otherwise a malicious connection can open a bunch of streams and send partial requests with 15.9MB headers and then pause, holding all the memory in place.

hyper will change default options for the safety of the majority, if you need to accept larger requests, you can set the option using the server builder.
